### PR TITLE
[DRAFT] add new p and q columns in _empty_res_switch

### PR DESCRIPTION
**Pull request by hilbrich**
adding switch p and q results

### CODE REVIEW

**Comment by vogt31337** [comment]
could you further explain why you want this patch to be included? Since switches normally don't have these values...

**Comment by vogt31337** [comment]
Okay, it's a WIP, nevermind.

File: pandapower/results_branch.py [hunk 1]
```diff
@@ -680,6 +680,20 @@ def _get_switch_results(net, i_ft, suffix=None):
         with np.errstate(invalid='ignore'):
             i_ka = np.max(i_ft[f:t], axis=1)
         net[res_switch_df].loc[net._impedance_bb_switches, "i_ka"] = i_ka
+
+        pf_mw = ppc["branch"][f:t, PF].real
```
Comment (rbolgaryn) [inline pandapower/results_branch.py:684]: I don't see a reason to have pf_mw and pt_mw if only p_from_mw and p_to_mw are used

File: pandapower/test/loadflow/test_results.py [hunk 2]
```diff
@@ -685,6 +685,34 @@ def test_bus_bus_switch(result_test_network, v_tol=1e-6, i_tol=1e-6, s_tol=5e-3,
...
+    # now test with some switches that have non-zero impedance:
+    net.switch.loc[[2,3], "z_ohm"] = 1e-3
+    pp.runpp(net)
+
+    p_from_ref = -0.29816
+    p_to_ref = 0.29816
+    q_from_ref = -0.96063
+    q_to_ref = 0.96063
+
+    assert isclose(net.res_switch.p_from_mw.at[2], p_from_ref, rtol=1e-3, atol=1e-6)
```
Comment (rbolgaryn) [inline pandapower/test/loadflow/test_results.py:702]: please set rtol to 0 and only use atol

File: pandapower/test/loadflow/test_results.py [hunk 3]
```diff
@@ -685,6 +685,34 @@ def test_bus_bus_switch(result_test_network, v_tol=1e-6, i_tol=1e-6, s_tol=5e-3,
...
+    assert isclose(net.res_switch.p_to_mw.at[2], p_to_ref, rtol=1e-3, atol=1e-6)
+    assert isclose(net.res_switch.q_from_mvar.at[2], q_from_ref, rtol=1e-3, atol=1e-6)
+    assert isclose(net.res_switch.q_to_mvar.at[2], q_to_ref, rtol=1e-3, atol=1e-6)
+
+    net.switch.loc[[2], "closed"] = False
+    pp.runpp(net)
+
+    assert isnan(net.res_switch.p_from_mw[2])
+
+    # return to normal test net with zero impedance switches:
```
Comment (rbolgaryn) [inline pandapower/test/loadflow/test_results.py:712]: if you are not testing anything after this it is not necessary to have this part

**Review by rbolgaryn — CHANGES_REQUESTED** [review]

File: pandapower/test/loadflow/test_results.py (hunk 2 above)
Comment (hilbrich) [reply pandapower/test/loadflow/test_results.py:702]: If I use rtol=0, how does it work? There is always a relative tolerance if I use an only approximately accurate value as a reference.

Comment (hilbrich) [reply pandapower/test/loadflow/test_results.py:702]: i have updated the code and used exact values, i just dont see the point to use isclose when using exact values. I changed the implementation by using exact values and therefore I will resolve this conversation, maybe we can discuss the usage of isclose later.

**Review by rbolgaryn — APPROVED** [review]

**Pull request merged**